lib/req_snowflake/req_snowflake.ex

```diff
@@ -0,0 +1,303 @@
+
+    parsed =
+      Task.async_stream(urls, fn url -> s3_get_json(url, key, md5) end, max_concurrency: 5)
+      |> Enum.map(fn {:ok, result} -> result end)
+      |> Enum.join(", ")
+
+    rows = Jason.decode!("[#{parsed}]")
+
+    row_data = process_row_data(rows, row_type)
+
+    columns = Enum.map(row_type, fn %{"name" => name} -> name end)
+
+    %Result{
+      success: true,
+      rows: row_data,
+      columns: columns,
+      num_rows: total
+    }
+  end
+
+  defp process_query_result_format(
+         "json",
+         %{
+           "rowset" => rowset,
+           "rowtype" => row_type,
+           "total" => total,
+           "chunks" => chunks,
+           "chunkHeaders" => %{
+             "x-amz-server-side-encryption-customer-key" => key,
+             "x-amz-server-side-encryption-customer-key-md5" => md5
+           }
+         }
+       ) do
+    parsed =
+      chunks
+      |> Enum.map(fn %{"url" => url} -> url end)
+      |> Task.async_stream(fn url -> s3_get_json(url, key, md5) end, max_concurrency: 5)
+      |> Enum.map(fn {:ok, result} -> result end)
+      |> Enum.join(", ")
+
+    rows = Jason.decode!("[#{parsed}]")
+
+    row_data = process_row_data(rowset, row_type) ++ process_row_data(rows, row_type)
+
+    columns = Enum.map(row_type, fn %{"name" => name} -> name end)
+
+    %Result{
+      success: true,
+      rows: row_data,
+      columns: columns,
+      num_rows: total
+    }
+  end
+
+  defp process_query_result_format(
+         "json",
+         %{"rowset" => rows, "rowtype" => row_type, "total" => total}
+       ) do
+    row_data = process_row_data(rows, row_type)
+
+    columns = Enum.map(row_type, fn %{"name" => name} -> name end)
+
+    %Result{
+      success: true,
+      rows: row_data,
+      columns: columns,
+      num_rows: total
+    }
+  end
+
+  defp snowflake_query_url(host) do
+    uuid = Application.get_env(:req_snowflake, :snowflake_uuid, UUID.uuid4())
+
+    "#{host}/queries/v1/query-request?requestId=#{uuid}"
+  end
+
+  defp snowflake_query_body(query) when is_binary(query) do
+    %{
+      sqlText: query,
+      sequenceId: 0,
+      bindings: nil,
+      bindStage: nil,
+      describeOnly: false,
+      parameters: %{
+        CLIENT_RESULT_CHUNK_SIZE: 48
+      },
+      describedJobId: nil,
+      isInternal: false,
+      asyncExec: false
+    }
+  end
+
+  defp snowflake_insert_headers(query, bindings) when is_binary(query) and is_map(bindings) do
+    %{
+      sqlText: query,
+      sequenceId: 0,
+      bindings: bindings,
+      bindStage: nil,
+      describeOnly: false,
+      parameters: %{
+        CLIENT_RESULT_CHUNK_SIZE: 48
+      },
+      describedJobId: nil,
+      isInternal: false,
+      asyncExec: false
+    }
+  end
+
+  defp s3_get_json(url, encryption_key, encryption_key_md5) do
+    Req.new(url: url)
+    |> Request.put_header("accept", "application/snowflake")
+    |> Request.put_header("Accept-Encoding", "gzip,deflate")
```

Comment:
no need for this since Req will add it for you by default (via compressed/1 step)

```suggestion
```